cmd/metal-api/internal/fsm/fsm_test.go

```diff
@@ -0,0 +1,270 @@
+					},
+				},
+				Liveliness: metal.MachineLivelinessAlive,
+			},
+			event: &metal.ProvisioningEvent{
+				Time:  now,
+				Event: metal.ProvisioningEventPhonedHome,
+			},
+			wantErr:            nil,
+			wantCrashLoop:      false,
+			wantFailedReclaim:  false,
+			wantLiveliness:     metal.MachineLivelinessAlive,
+			wantNumberOfEvents: 1,
+			wantLastEventTime:  now,
+			wantLastEvent:      metal.ProvisioningEventPhonedHome.String(),
+		},
+		{
+			name: "Swallow repeated Phoned Home",
+			container: &metal.ProvisioningEventContainer{
+				Events: metal.ProvisioningEvents{
+					{
+						Time:  lastTimeEvent,
+						Event: metal.ProvisioningEventPhonedHome,
+					},
+				},
+				Liveliness: metal.MachineLivelinessAlive,
+			},
+			event: &metal.ProvisioningEvent{
+				Time:  now,
+				Event: metal.ProvisioningEventPhonedHome,
+			},
+			wantErr:            nil,
+			wantCrashLoop:      false,
+			wantFailedReclaim:  false,
+			wantLiveliness:     metal.MachineLivelinessAlive,
+			wantNumberOfEvents: 1,
+			wantLastEventTime:  now,
+			wantLastEvent:      metal.ProvisioningEventPhonedHome.String(),
+		},
+		{
+			name: "Swallow Phoned Home after Machine Reclaim",
+			container: &metal.ProvisioningEventContainer{
+				Events: metal.ProvisioningEvents{
+					{
+						Time:  lastTimeEvent,
+						Event: metal.ProvisioningEventMachineReclaim,
+					},
+				},
+				Liveliness:    metal.MachineLivelinessAlive,
+				LastEventTime: &lastTimeEvent,
+			},
+			event: &metal.ProvisioningEvent{
+				Time:  now,
+				Event: metal.ProvisioningEventPhonedHome,
+			},
+			wantErr:            nil,
+			wantCrashLoop:      false,
+			wantFailedReclaim:  false,
+			wantLiveliness:     metal.MachineLivelinessAlive,
+			wantNumberOfEvents: 1,
+			wantLastEventTime:  lastTimeEvent,
+			wantLastEvent:      metal.ProvisioningEventMachineReclaim.String(),
+		},
+		{
+			name: "Failed Machine Reclaim",
+			container: &metal.ProvisioningEventContainer{
+				Events: metal.ProvisioningEvents{
+					{
+						Time:  lastTimeEvent,
+						Event: metal.ProvisioningEventMachineReclaim,
+					},
+				},
+				Liveliness:    metal.MachineLivelinessAlive,
+				LastEventTime: &lastTimeEvent,
+			},
+			event: &metal.ProvisioningEvent{
+				Time:  now.Add(time.Minute * 10),
+				Event: metal.ProvisioningEventPhonedHome,
+			},
+			wantErr:            nil,
+			wantCrashLoop:      false,
+			wantFailedReclaim:  true,
+			wantLiveliness:     metal.MachineLivelinessAlive,
+			wantNumberOfEvents: 1,
+			wantLastEventTime:  now.Add(time.Minute * 10),
+			wantLastEvent:      metal.ProvisioningEventMachineReclaim.String(),
+		},
+	}
+	for _, tt := range tests {
+		log := zap.NewExample().Sugar()
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := handleProvisioningEvent(tt.event, tt.container, log)
+			if diff := cmp.Diff(tt.wantErr, err); diff != "" {
+				t.Errorf("HandleProvisioningEvent() diff = %s", diff)
+			}
+
+			if tt.container.CrashLoop != tt.wantCrashLoop {
```

Comment:
Why do you compare against `tt.container.CrashLoop` and not the event container returned by `handleProvisioningEvent`?

You can also do this check in one-line: `assert.Equal(t, tt.wantCrashLoop, got.CrashLoop)`